src/plugins/fakeNitro.tsx

```diff
@@ -344,26 +430,125 @@ export default definePlugin({
                 continue;
             }
 
-            const fakeNitroMatch = element.props.href.match(/https:\/\/cdn\.discordapp\.com\/emojis\/(\d+?)\.(png|webp|gif).+?(?=\s|$)/);
-            if (!fakeNitroMatch) {
-                newContent.push(element);
-                continue;
+            if (settings.store.transformEmojis) {
+                const fakeNitroMatch = element.props.href.match(fakeNitroEmojiRegex);
+                if (fakeNitroMatch) {
+                    const emojiName = EmojiStore.getCustomEmojiById(fakeNitroMatch[1])?.name ?? element.props.href.match(fakeNitroNameRegex)?.[1] ?? "FakeNitroEmoji";
+
+                    newContent.push(MessageElementsParser.customEmoji.react({
```

Comment:
the way u get this is really cursed, you can just use the Parser from webpack commons

```suggestion
                    newContent.push(Parser.defaultRules.customEmoji.react({
```